File: WorldFeed/Server/src/WorldFeed.Common/Credentials/ICredentialsAccessor.cs

namespace WorldFeed.Common.Credentials
{
    using System;

    using WorldFeed.Common.Public.Models.Authentication;

    public interface ICredentialsAccessor
    {
        ITwitterCredentials ApplicationCredentials { get; set; }

        ITwitterCredentials CurrentThreadCredentials { get; set; }

        T ExecuteOperationWithCredentials<T>(ITwitterCredentials credentials, Func<T> operation);

        void ExecuteOperationWithCredentials(ITwitterCredentials credentials, Action operation);
    }
}

File: WorldFeed/Client/src/app/core/Core/Credentials/ICredentialsAccessor.ts
import {ITwitterCredentials} from "../../Public/Models/Authentication/TwitterCredentials";

export interface ICredentialsAccessor {
  applicationCredentials: ITwitterCredentials;
  currentThreadCredentials: ITwitterCredentials;

  executeOperationWithCredentials<T>(credentials: ITwitterCredentials, operation: () => T): T;

  executeOperationWithCredentials(credentials: ITwitterCredentials, operation: () => void): void;
}
